Psychtoolbox/PsychHardware/EyelinkToolbox/EyelinkDemos/SR-ResearchDemos/SimpleVideo/EyeLink_SimpleVideo.m

```diff
@@ -0,0 +1,396 @@
+        Eyelink('Command', 'file_sample_data  = LEFT,RIGHT,GAZE,HREF,RAW,AREA,HTARGET,GAZERES,BUTTON,STATUS,INPUT');
+        Eyelink('Command', 'link_sample_data  = LEFT,RIGHT,GAZE,GAZERES,AREA,HTARGET,STATUS,INPUT');
+    else
+        Eyelink('Command', 'file_sample_data  = LEFT,RIGHT,GAZE,HREF,RAW,AREA,GAZERES,BUTTON,STATUS,INPUT');
+        Eyelink('Command', 'link_sample_data  = LEFT,RIGHT,GAZE,GAZERES,AREA,STATUS,INPUT');
+    end
+    
+    
+    %% STEP 3: OPEN GRAPHICS WINDOW
+    
+    % Open experiment graphics on the specified screen
+    if isempty(screenNumber)
+        screenNumber = max(Screen('Screens')); % Use default screen if none specified
+    end
+    window = Screen('OpenWindow', screenNumber, [128 128 128]); % Open graphics window
+    Screen('Flip', window);
+    % Return width and height of the graphics window/screen in pixels
+    [width, height] = Screen('WindowSize', window);
+    
+    
+    %% STEP 4: SET CALIBRATION SCREEN COLOURS/SOUNDS; PROVIDE WINDOW SIZE TO EYELINK HOST & DATAVIEWER; SET CALIBRATION PARAMETERS; CALIBRATE
+    
+    % Provide EyeLink with some defaults, which are returned in the structure "el".
+    el = EyelinkInitDefaults(window);
+    % set calibration/validation/drift-check(or drift-correct) background color. 
+    % It is important that this background colour is similar to that of the stimuli to prevent large luminance-based 
+    % pupil size changes (which can cause a drift in the eye movement data)
+    el.backgroundcolour = [115 115 115];% RGB grey
+    % set "Camera Setup" instructions text colour so it is different from background colour
+    el.msgfontcolour = [0 0 0];% RGB black
+    
+    % Set calibration beeps (0 = sound off, 1 = sound on)
+    % Setting beeps to off (0) for video targets
+    el.targetbeep = 0;  % sound a beep when a target is presented
+    el.feedbackbeep = 0;  % sound a beep after calibration or drift check/correction
+    
+    % Required for macOS Catalina users (/w PTB 3.0.17.11) to disable audio
+    % with animated calibration targets and trial video stimuli to
+    % avoid freezing in video playback
+    spcf1 = 0; % Used both here in el struct for animated target movies, and also below when loading trial-stimulus movies
+    if IsOSX
+        [status, result] = system('sw_vers');
+        if regexp(result,'ProductVersion\D*10\.15')
+            spcf1 = 2;
+            el.calAnimationOpenSpecialFlags1 = spcf1; % for Screen('OpenMovie', ..., specialFlags1) see http://psychtoolbox.org/docs/Screen-OpenMovie
+        end
+    end
+    % N.B. el.calAnimationOpenSpecialFlags1 = 2 may also be required with certain versions of 
+    % GStreamer on Ubuntu Linux. In certain cases on Linux (e.g. when PsychPortAudio('GetOpenDeviceCount') > 0), 
+    % disabling the audio is required to avoid a crash, or more generally a brief delay oafter the first 
+    % frame of the target video is displayed on screen and the eventual continuation of playback

```

Comment:
This is probably due to a Pulseaudio version 16 sound server bug in Ubuntu 22.04 - 23.10. See https://github.com/Psychtoolbox-3/Psychtoolbox-3/issues/814 for details and possible workarounds at the end, if low audio latency during playback of the movies soundtrack is not required.

Hopefully this will be fixed for a future Ubuntu 24.04-LTS next spring.

Reply:
will keep an eye on this in next few months